### DISCUSSION TITLE
Manual fixes for facial recognition (identical twins)

### DISCUSSION THREAD

**Discussion by natefelt** [immich-app/immich#3946] [Q&A] [answered]
I realize this is a very niche use case and won't affect many people. It affects all photo library software I've tried with facial recognition to some degree. I've just started with immich so forgive me if there is an obvious answer. 

Facial recognition generally works well, but unsurprisingly, my identical twins are "tagged" as the same person. In photos of the two of them together, only one person is listed. Photos of them by their self are grouped under the twin I first tagged.  Is there a way to manually tag a second person or change who was auto-detected?

**Comment by mertalev** [comment]
If you go to the admin settings, there's a setting under facial recognition for "max recognition distance". The default is 0.6; decreasing this a bit can make it distinguish between them. I'd start with 0.5, re-run facial recognition and go from there.

**Reply by natefelt** [reply]
This did help a bit, but I realize facial recognition in immich is still young and I'll just need to be patient for updates.

**Reply by mertalev** [reply, marked as answer]
You can decrease it even further, but too low and it will start separating the same person into different people. We're hoping to improve the clustering algorithm for facial recognition, which will make these cases more rare.

**Comment by maltokyo** [comment]
I also have a few "edge cases" that I would like to fix manually (split one person into two, un-tag a face that was recognized by mistake etc).  Is there a way to manually do this, that also helps the algorithm "learn" so it does not repeat the same mistake in the future?

**Reply by maltokyo** [reply]
@mertalev - to clarify the above, I think it would be helpful for users to keep the default setting, but fix some cases of similar faces manually, so that the algorithm would learn for those particular faces, and maybe just reduce the setting for them specifically.  Not sure if that is how it works - in my case, I have two pairs of faces that are always classified as the same person when they are not - I don't want to change the algorithm for any of the other people where it is working fine.

**Reply by mertalev** [reply]
Splitting one person in two is currently being worked on, see #4153 and #4717). I think the current facial clustering algorithm will perform much better at differentiating between the two after splitting them with this feature.

**Comment by RaniaBZ** [comment]
I'm one of the triplets and want to solve this problem, will anyone help me?

**Comment by DanielKinsman** [comment]
It's funny, many of the "normal" people in my library were recognised as separate people, yet the hit rate on me and my brother being the same person is 100%.

**Comment by pabera** [comment]
I case re-running the face recognition is not working for you (it takes too much time, time spend on naming and merging would be lost), I have come up with a small tool to fix face assignments manually and partially automatically.

* https://github.com/pabera/immich-face-fix
